src/wallet/init.cpp

```diff
@@ -129,5 +130,7 @@ void WalletInit::Construct(NodeContext& node) const
             settings.rw_settings["wallet"] = wallets;
         });
     }
-    node.chain_clients.emplace_back(interfaces::MakeWalletClient(*node.chain, args, args.GetArgs("-wallet")));
+    auto wallet_client = interfaces::MakeWalletClient(*node.chain, args, args.GetArgs("-wallet"));
+    node.wallet_client = wallet_client.get();
```

Comment:
nit in e4f435047121886edb6e6a6c4e4998e44ed2e36a:

Would be good to assert `!node.wallet_client` to avoid accidentally re-seating the reference